handlers/practitioner_resource.go

```diff
@@ -78,22 +85,15 @@ func HandleCreatePractitionersResource(svc dao.Service, helperService utils.Help
 		}
 
 		practitionerDao := transformers.PractitionerResourceRequestToDB(&request, practitionerID, transactionID)
-		practitionerDao.Etag = etag
-		practitionerDao.Kind = "insolvency#practitioner"
-		practitionerResourceDto.Data = *practitionerDao
-		practitionerResourceDto.Data.PractitionerId = practitionerID
-
-		// GetInsolvencyResourceData retrieves previously stored practitioners
-		insolvencyResourceDaoDataDto, err := svc.GetInsolvencyResourceData(transactionID)
-		if err != nil {
-			logErrorAndHttpResponse(w, req, http.StatusInternalServerError, "error", []error{err})
-			return
-		}
+		practitionerDao.Data.Etag = etag
+		practitionerDao.Data.Kind = "insolvency#practitioner"
+		practitionerResourceDao = *practitionerDao
+		practitionerResourceDao.Data.PractitionerId = practitionerID
 
 		maxPractitioners := 5
 		//check to ensure it is not nil from the collection
-		if insolvencyResourceDaoDataDto != nil && len(insolvencyResourceDaoDataDto.Practitioners) > 0 {
-			err = json.Unmarshal([]byte(insolvencyResourceDaoDataDto.Practitioners), &practitionersMapResource)
+		if insolvencyResourceDao != nil && len(insolvencyResourceDao.Data.Practitioners) > 0 {
```

Comment:
I'm not sure of the purpose of this check for nil - if to protect against a null pointer, then a check on .Data may need to be included (also in line 178)

Reply:
They are 2 different necessary checks to be honest. if len(nil) for instance my code would panic,that necessitates checks for nil before Data.practitioners. if first is not nil, there must be Data and if it has empty Practitioners,the code would not panic.. The point I'm making is that both are good checks. Also if I'm wrong, golang code linting would have told me so as well.Thanks

Reply:
"if first is not nil, there must be Data" - this was the part I was asking about - makes sense that if the struct exists as defined in the model, it has a Data field, thanks.